lib/rets/client.rb

```diff
@@ -278,11 +278,28 @@ def metadata
     def retrieve_metadata
       raw_metadata = {}
       Metadata::METADATA_TYPES.each {|type|
-        raw_metadata[type] = retrieve_metadata_type(type)
+        raw_metadata[type] = filter_metadata(type, retrieve_metadata_type(type))
       }
       raw_metadata
     end
 
+    def filter_metadata(metadata_type, xml)
+      return xml unless @options[:metadata_filter]
+
+      types_to_filter = @options[:metadata_filter][:types_to_filter]
```

Comment:
right, that makes sense. Can that just be the default rather than requiring each mls to specify which types to filter?